x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java

```diff
@@ -786,6 +789,117 @@ && isMillisOrNanos(listOfTypes.get(j))) {
         }
     }
 
+    public void testDateMathIndexPattern() throws IOException {
+        ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
```

Comment:
Using now may lead to flakiness when the tests are executed around midnight, no? Maybe a fixed date would be better?

Reply:
I thought about it, but date math is all about dynamic patterns using `now`. I'm not sure a test with a fixed date will be realistic.

Reply:
How long does this test runs? 
I have seen some sql/eql examples that add assumption they are executed at least 10-15 minutes before midnight.